### PR TITLE
Fixed more warnings

### PR DESCRIPTION
**Pull request by alexban011**
Last report: [reports.zip](https://github.com/yairm210/Unciv/files/11502648/reports.zip)

### CODE REVIEW

**Comment by yairm210** [comment]
Conflicts

File: core/src/com/unciv/logic/BackwardCompatibility.kt [hunk 1]
```diff
@@ -156,9 +156,8 @@ object BackwardCompatibility {
     }
 
     /** Move max XP from barbarians to new home */
-    @Suppress("DEPRECATION")
-    fun ModOptions.updateDeprecations() {
-    }
+    @Suppress("DEPRECATION", "EmptyFunctionBlock")
+    fun ModOptions.updateDeprecations() { }
```
Comment (alexban011) [inline core/src/com/unciv/logic/BackwardCompatibility.kt:160]: Why exactly are we calling this function?

File: tests/src/com/unciv/dev/FasterUIDevelopment.kt [hunk 2]
```diff
@@ -41,6 +41,7 @@ object FasterUIDevelopment {
             actor = "This could be your UI element in development!".toLabel()
         }
 
+        @Suppress("EmptyFunctionBlock")
         fun afterAdd() {
```
Comment (alexban011) [inline tests/src/com/unciv/dev/FasterUIDevelopment.kt:45]: same here

File: core/src/com/unciv/ui/screens/victoryscreen/ReplayMap.kt [hunk 3]
```diff
@@ -47,6 +47,7 @@ abstract class IndependentMiniMap(
     protected open fun includeTileFilter(tile: Tile): Boolean = true
 
     private fun createReplayMap(tileSize: Float): List<MinimapTile> {
+        @Suppress("EmptyFunctionBlock")
         val doNothing = fun(){}
```
Comment (alexban011) [inline core/src/com/unciv/ui/screens/victoryscreen/ReplayMap.kt:51]: same here. MinimapTile(tile, tileSize, doNothing) could just be instantiated without doNothing right?

File: core/src/com/unciv/logic/civilization/ExploredRegion.kt [hunk 4]
```diff
@@ -12,7 +12,7 @@ import com.unciv.ui.components.tilegroups.TileGroupMap
 import kotlin.math.abs
 import kotlin.math.sqrt
 
-class ExploredRegion () : IsPartOfGameInfoSerialization {
```
Comment (alexban011) [inline core/src/com/unciv/logic/civilization/ExploredRegion.kt:15]: is the default costructor neccessary for serialization? Should I suppress the warning?

**Comment by alexban011** [comment]
And finally:
> Disabled TooGenericExceptionCaught && TooGenericExceptionThrown

Should I disable these? There are quite a few cases where we want to do this (like X threw an exception -> fall back to whatever) so the code would be full of @Suppress annotations

File: core/src/com/unciv/logic/BackwardCompatibility.kt (hunk 1 above)
Comment (yairm210) [reply core/src/com/unciv/logic/BackwardCompatibility.kt:160]: This is so if there's stuff to deprecate the devs can add the stages here. Currently there are none, but this is legit

File: tests/src/com/unciv/dev/FasterUIDevelopment.kt (hunk 2 above)
Comment (yairm210) [reply tests/src/com/unciv/dev/FasterUIDevelopment.kt:45]: This is a dev helper tool, this is legit

**Comment by yairm210** [comment]
And yes, keep those disabled

All in all excellent work!

File: core/src/com/unciv/logic/civilization/ExploredRegion.kt (hunk 4 above)
Comment (alexban011) [reply core/src/com/unciv/logic/civilization/ExploredRegion.kt:15]: thanks!

**Pull request merged**